Lib/json/encoder.py

```diff
@@ -350,7 +350,10 @@ def _iterencode_dict(dct, _current_indent_level):
             item_separator = _item_separator
         first = True
         if _sort_keys:
-            items = sorted(dct.items())
+            try:
+                items = sorted(dct.items())
+            except(TypeError):
+                items = sorted(dct.items(), key=lambda kv: str(kv[0]))
```

Comment:
It does not work correctly for True, False, None and some int and float subclasses (like IntEnum).

Reply:
Yes, indeed it doesn't work. I have no idea how to correct it.